src/MICore/LinuxNativeMethods.cs

```diff
@@ -13,8 +13,8 @@ internal class LinuxNativeMethods
         [DllImport(Libc, EntryPoint = "kill", SetLastError = true)]
         internal static extern int Kill(int pid, int mode);
 
-        [DllImport(Libc, EntryPoint = "mkfifo", SetLastError = true)]
-        internal static extern int MkFifo(string name, int mode);
+        [DllImport(Libc, EntryPoint = "mkfifo", SetLastError = true, ThrowOnUnmappableChar = true)]
+        internal static extern int MkFifo([MarshalAs(UnmanagedType.LPWStr)]string name, int mode);
```

Comment:
Did you actually test this marshalling on Linux? The Unicode mappings there behave differently than windows. Not sure if this marshalas is correct. Furthermore, I think this fxcop error is actually bogus off windows since there aren't A and W versions of the functions. Is the ThrowOnUnmappableChar warning actually in the divisional min bar?